torchvision/prototype/transforms/_geometry.py

```diff
@@ -92,6 +92,9 @@ def _get_params(self, sample: Any) -> Dict[str, Any]:
         # vfdev-5: techically, this op can work on bboxes/segm masks only inputs without image in samples
         # What if we have multiple images/bboxes/masks of different sizes ?
         # TODO: let's support bbox or mask in samples without image
+        # datumbox: One issue with implementing the above proposal is that `query_chw()` is supposed to return channels
+        # that bboxes and masks don't have. It could return None, as the majority of ops don't need it. Thoughts?
+        # The same applies to all similar TODOs in this file.
```

Comment:
Not something we need to do now, just getting your thoughts @vfdev-5 / @pmeier

Reply:
My original proposal from way back was to split this into two functions, i.e. `query_hw` and `query_c`. This way the former would work for all types and the latter only for images.

Given that we need to go through the sample twice and we do that quite often, a better option might be return `c=None` as you suggested. Still, I would at least add a `channels_needed: bool = False` flag. If true, you are safe to never get `None`. This way we don't need extra error handling in every transformation that actually needs the number of channels.

Reply:
OK, we can do what you proposed (split). We don't need to do this now as this is not critical for the API finalization. I think to mark this comment as resolved, you can just open an issue with your proposal (so that we don't forget) and comment the id here.

Reply:
See #6491.